vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/RecalculateColumnWidthsPage.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package com.vaadin.flow.component.grid.it;
+
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.router.Route;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+@Route("vaadin-grid/recalculate-column-widths")
+public class RecalculateColumnWidthsPage extends VerticalLayout {
+
+    public RecalculateColumnWidthsPage() {
+        ThreeString ts1 = new ThreeString("111", "222", "333");
+        ThreeString ts2 = new ThreeString("444", "555", "667");
+
+        Collection<ThreeString> itemList = new ArrayList<>();
+        itemList.add(ts1);
+        itemList.add(ts2);
+
+        Grid<ThreeString> grid1 = new Grid<>();
+        grid1.setId("grid");
+        grid1.setHeightByRows(true);
+        grid1.setItems(itemList);
+
+        grid1.addColumn(ThreeString::getA).setAutoWidth(true);
+        grid1.addColumn(ThreeString::getB).setAutoWidth(true);
+        grid1.addColumn(ThreeString::getC).setAutoWidth(true);
```

Comment:
You access the fields directly and remove the getters and setters in the bean to reduce the amount of code.
```suggestion
        grid1.addColumn(item -> item.a).setAutoWidth(true);
        grid1.addColumn(item -> item.b).setAutoWidth(true);
        grid1.addColumn(item -> item.c).setAutoWidth(true);
```